packages/dds/tree/src/feature-libraries/forest-summary/forestSummarizer.ts

```diff
@@ -89,32 +131,51 @@ export class ForestSummarizer implements Summarizable {
 			);
 			fieldMap.set(key, innerCursor as ITreeCursorSynchronous & ITreeSubscriptionCursor);
 		});
-		const encoded = this.codec.encode(fieldMap, this.encoderContext);
 
+		const forestSummaryBuilder = new SummaryTreeBuilder();
+		// Let the incremental summary builder know that we are starting a new summary.
+		// It returns whether incremental encoding is enabled.
+		const shouldEncodeIncrementally = this.incrementalSummaryBuilder.startingSummary(
+			forestSummaryBuilder,
+			fullTree,
+			incrementalSummaryContext,
+		);
+		const encoderContext: FieldBatchEncodingContext = {
+			...this.encoderContext,
+			incrementalEncoderDecoder: shouldEncodeIncrementally
+				? this.incrementalSummaryBuilder
+				: undefined,
+		};
+		const encoded = this.codec.encode(fieldMap, encoderContext);
 		fieldMap.forEach((value) => value.free());
-		return stringify(encoded);
-	}
 
-	public summarize(props: {
-		stringify: SummaryElementStringifier;
-		fullTree?: boolean;
-		trackState?: boolean;
-		telemetryContext?: ITelemetryContext;
-		incrementalSummaryContext?: IExperimentalIncrementalSummaryContext;
-	}): ISummaryTreeWithStats {
-		return createSingleBlobSummary(treeBlobKey, this.getTreeString(props.stringify));
+		forestSummaryBuilder.addBlob(forestSummaryContentKey, stringify(encoded));
+		// Let the incremental summary builder know that we are done with this summary.
+		this.incrementalSummaryBuilder.completedSummary(incrementalSummaryContext);
+		return forestSummaryBuilder.getSummaryTree();
 	}
 
 	public async load(
 		services: IChannelStorageService,
 		parse: SummaryElementParser,
 	): Promise<void> {
-		if (await services.contains(treeBlobKey)) {
-			const treeBuffer = await services.readBlob(treeBlobKey);
-			const treeBufferString = bufferToString(treeBuffer, "utf8");
+		if (await services.contains(forestSummaryContentKey)) {
+			const readAndParse: ReadAndParseBlob = async <T>(id: string): Promise<T> => {
+				const blob = await services.readBlob(id);
+				const decoded = bufferToString(blob, "utf8");
```

Comment:
I find renaming `treeBufferString` to `decoded` to be counterproductive.

We have ForestCodec which decodes things, and we have utf8 which is used to decode things, and we have the fluid serializer which we use to decode strings into json like objects with handles. If I understand correctly, this has undergone 1 of 3 decode operations. Simply calling it "decoded" doesn't seem like a good way to communicate that.

Reply:
Reverted the names.